presto-accumulo/src/main/java/com/facebook/presto/accumulo/AccumuloClient.java

```diff
@@ -0,0 +1,1160 @@
+
+        // Create the AccumuloTable object
+        AccumuloTable table = new AccumuloTable(meta.getTable().getSchemaName(),
+                meta.getTable().getTableName(), columns, rowIdColumn,
+                AccumuloTableProperties.isExternal(tableProperties),
+                AccumuloTableProperties.getSerializerClass(tableProperties),
+                AccumuloTableProperties.getScanAuthorizations(tableProperties));
+
+        // First, create the metadata
+        metaManager.createTableMetadata(table);
+
+        // Make sure the namespace exists
+        tableManager.ensureNamespace(table.getSchema());
+
+        // Create the Accumulo table if it does not exist (for 'external' table)
+        if (!tableManager.exists(table.getFullTableName())) {
+            tableManager.createAccumuloTable(table.getFullTableName());
+        }
+
+        // Set any locality groups on the data table
+        setLocalityGroups(tableProperties, table);
+
+        // Create index tables, if appropriate
+        createIndexTables(table);
+
+        return table;
+    }
+
+    /**
+     * Validates the given metadata for a series of conditions to ensure the table is well-formed
+     *
+     * @param meta Table metadata
+     */
+    private void validateCreateTable(ConnectorTableMetadata meta)
+    {
+        validateColumns(meta);
+        validateLocalityGroups(meta);
+        if (!AccumuloTableProperties.isExternal(meta.getProperties())) {
+            validateInternalTable(meta);
+        }
+    }
+
+    /**
+     * Validates the columns set in the configure locality groups (if any) exist
+     *
+     * @param meta Table metadata
+     */
+    private void validateColumns(ConnectorTableMetadata meta)
+    {
+        // Here, we make sure the user has specified at least one non-row ID column
+        // Accumulo requires a column family and qualifier against a row ID, and these values
+        // are specified by the other columns in the table
+        if (meta.getColumns().size() == 1) {
+            throw new InvalidParameterException("Must have at least one non-row ID column");
+        }
+
+        // Check all the column types, and throw an exception if the types of a map are complex
+        // While it is a rare case, this is not supported by the Accumulo connector
+        ImmutableSet.Builder<String> columnNameBuilder = ImmutableSet.builder();
+        for (ColumnMetadata column : meta.getColumns()) {
+            if (Types.isMapType(column.getType())) {
+                if (Types.isMapType(Types.getKeyType(column.getType()))
+                        || Types.isMapType(Types.getValueType(column.getType()))
+                        || Types.isArrayType(Types.getKeyType(column.getType()))
+                        || Types.isArrayType(Types.getValueType(column.getType()))) {
+                    throw new PrestoException(VALIDATION,
+                            "Key/value types of a MAP column must be plain types");
```

Comment:
Maps of complex types are not supported, though multi-dimensional arrays are.  However, I am not checking the case when a multi-dimensional array of maps with complex types is used.  This case should be checked as well.